### PR TITLE
[Provisioner] Get rid of ray dependency locally for aws

### PR DESCRIPTION
**Pull request by Michaelvll**
Follow up for #1702 to get rid of ray dependency locally for AWS

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
  - In a new docker container `docker run -it --rm --name skypilot-debug berkeleyskypilot/skypilot-debug /bin/bash`
  - [x] `time pip install -e .`:  28s
  - [x] `sky launch --cloud aws --cpus 2 -c test-launch-aws`
  - [x] `time pip install -e .[gcp]`: 25s ; `time conda install -c conda-forge google-cloud-sdk`  (1m20s)
  - [x] `sky launch --cloud aws --cpus 2 -c test-launch-aws-with-ray`
  - [x] `sky launch --cloud gcp --cpus 2 -c test-launch-gcp`
  - [ ] `pip install -e .[lambda]`; `sky launch --cloud lambda --cpus 2+ -c test-lambda` (failed to get capacity)
  - [x] In a new container, `pip install -e .[all]`: 2m30s (originally: 2m30s); `sky launch --cloud aws --cpus 2 -c test-launch-aws`
- [ ] All smoke tests: `pytest tests/test_smoke.py`
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/setup_files/setup.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def parse_readme(readme: str) -> str:
 extras_require: Dict[str, List[str]] = {
     # TODO(zwhu): remove the local_ray requirement once we resolve the
     # non-critical dependency issues in #2625
```
Comment (concretevitamin) [inline sky/setup_files/setup.py:133]: remove?

File: sky/clouds/utils/__init__.py
Comment (concretevitamin) [inline sky/clouds/utils/__init__.py:1]: What's the reason of moving these `<cloud>_utils.py` to a new module?

We may want to document in README.md in this dir to give future clouds some guidance.

**Review by concretevitamin — COMMENTED** [review]
Thanks @Michaelvll, a quick question.

Also, previously there was some logging change, is that no longer needed?

File: sky/clouds/utils/__init__.py
Comment (suquark) [reply sky/clouds/utils/__init__.py:1]: +1

File: sky/clouds/utils/scp_utils.py [hunk 2]
```diff
@@ -222,88 +223,88 @@ def _delete(self, url, request_body=None):
...
             }, {
-                "serviceType": "TCP",
-                "serviceValue": "443"
+                'serviceType': 'TCP',
+                'serviceValue': '443'
             }],
-            "destinationIpAddresses": ["0.0.0.0/0"],
-            "ruleDescription": "skypilot out rue"
+            'destinationIpAddresses': ['0.0.0.0/0'],
+            'ruleDescription': 'skypilot out rue'
```
Comment (suquark) [inline sky/clouds/utils/scp_utils.py:264]: Is "rue" a typo?

Comment (Michaelvll) [reply sky/clouds/utils/scp_utils.py]: Fixed. Thanks!

File: sky/clouds/utils/__init__.py
Comment (Michaelvll) [reply sky/clouds/utils/__init__.py:1]: Added a simple one. We may want to add a more detailed one to the new cloud guide instead?

**Review by concretevitamin — APPROVED** [review]
LGTM, thanks @Michaelvll.

**Pull request merged**